corehq/apps/app_manager/models.py

```diff
@@ -2291,10 +2334,12 @@ def add_insert_form(self, from_module, form, index=None, with_source=False):
                             case_type=module.case_type,
                             case_tag='_'.join(['parent'] * (i + 1)),
                             details_module=module.unique_id,
-                            parent_tag='_'.join(['parent'] * (i + 2)) if n > 0 else ''
+                            parents=[
+                                ParentIndex(tag='_'.join(['parent'] * (i + 2)) if n > 0 else '')
```

Comment:
Since you've added the `relationship` field to basic modules you should probably  copy that over as well.